tools/tpm2_createek.c

```diff
@@ -178,6 +209,89 @@ static tool_rc set_ek_template(ESYS_CONTEXT *ectx, TPM2B_PUBLIC *input_public) {
     return rc;
 }
 
+static void set_default_obj_attrs(void) {
+
+    ctx.objdata.in.public.publicArea.objectAttributes =
+      TPMA_OBJECT_RESTRICTED  | TPMA_OBJECT_ADMINWITHPOLICY
+    | TPMA_OBJECT_DECRYPT     | TPMA_OBJECT_FIXEDTPM
+    | TPMA_OBJECT_FIXEDPARENT | TPMA_OBJECT_SENSITIVEDATAORIGIN;
+
+    if (ctx.objdata.in.public.publicArea.nameAlg != TPM2_ALG_SHA256)
+    {
+       ctx.objdata.in.public.publicArea.objectAttributes |=
+            TPMA_OBJECT_USERWITHAUTH;
+    }
+}
+
+static void set_default_auth_policy(void) {
+
+    static const TPM2B_DIGEST auth_policy_A_SHA256 = {
+        .size = 32,
+        .buffer = {
+            0x83, 0x71, 0x97, 0x67, 0x44, 0x84, 0xB3, 0xF8, 0x1A, 0x90,
+            0xCC, 0x8D, 0x46, 0xA5, 0xD7, 0x24, 0xFD, 0x52, 0xD7, 0x6E,
+            0x06, 0x52, 0x0B, 0x64, 0xF2, 0xA1, 0xDA, 0x1B, 0x33, 0x14,
+            0x69, 0xAA
+        }
+    };
+
+    static const TPM2B_DIGEST auth_policy_B_SHA384 = {
+        .size = 48,
+        .buffer = {
+            0xB2, 0x6E, 0x7D, 0x28, 0xD1, 0x1A, 0x50, 0xBC, 0x53, 0xD8,
+            0x82, 0xBC, 0xF5, 0xFD, 0x3A, 0x1A, 0x07, 0x41, 0x48, 0xBB,
+            0x35, 0xD3, 0xB4, 0xE4, 0xCB, 0x1C, 0x0A, 0xD9, 0xBD, 0xE4,
+            0x19, 0xCA, 0xCB, 0x47, 0xBA, 0x09, 0x69, 0x96, 0x46, 0x15,
+            0x0F, 0x9F, 0xC0, 0x00, 0xF3, 0xF8, 0x0E, 0x12
+        }
+    };
+
+    static const TPM2B_DIGEST auth_policy_B_SHA512 = {
+        .size = 64,
+        .buffer = {
+            0xB8, 0x22, 0x1C, 0xA6, 0x9E, 0x85, 0x50, 0xA4, 0x91, 0x4D,
+            0xE3, 0xFA, 0xA6, 0xA1, 0x8C, 0x07, 0x2C, 0xC0, 0x12, 0x08,
+            0x07, 0x3A, 0x92, 0x8D, 0x5D, 0x66, 0xD5, 0x9E, 0xF7, 0x9E,
+            0x49, 0xA4, 0x29, 0xC4, 0x1A, 0x6B, 0x26, 0x95, 0x71, 0xD5,
+            0x7E, 0xDB, 0x25, 0xFB, 0xDB, 0x18, 0x38, 0x42, 0x56, 0x08,
+            0xB4, 0x13, 0xCD, 0x61, 0x6A, 0x5F, 0x6D, 0xB5, 0xB6, 0x07,
+            0x1A, 0xF9, 0x9B, 0xEA
+        }
+    };
+
+    static const TPM2B_DIGEST auth_policy_B_SM3_256 = {
+        .size = 32,
+        .buffer = {
+            0x16, 0x78, 0x60, 0xA3, 0x5F, 0x2C, 0x5C, 0x35, 0x67, 0xF9,
+            0xC9, 0x27, 0xAC, 0x56, 0xC0, 0x32, 0xF3, 0xB3, 0xA6, 0x46,
+            0x2F, 0x8D, 0x03, 0x79, 0x98, 0xE7, 0xA1, 0x0F, 0x77, 0xFA,
+            0x45, 0x4A
+        }
+    };
+
+    TPM2B_DIGEST *authp = &ctx.objdata.in.public.publicArea.authPolicy;
+
+    switch (ctx.objdata.in.public.publicArea.nameAlg) {
+    case TPM2_ALG_SHA256:
```

Comment:
You need a falls-thru comment here to keep some compilers happy.

Reply:
I have never met such compilers but I dropped the line anyway.

Reply:
https://developers.redhat.com/blog/2017/03/10/wimplicit-fallthrough-in-gcc-7

Reply:
Actually there is no fall-thru in the switch statement above. `TPM2B_DIGEST *authp = ...` is a pointer definition outside the switch statement.